dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/TrackedEntityAttributeControllerTest.java

```diff
@@ -0,0 +1,231 @@
+
+    private TrackedEntityAttribute teaE;
+
+    @BeforeEach
+    void setUp()
+    {
+        orgUnit = createOrganisationUnit( 'A' );
+        manager.save( orgUnit );
+
+        program = createProgram( 'A' );
+        manager.save( program );
+
+        teaA = createTrackedEntityAttribute( 'A' );
+        teaB = createTrackedEntityAttribute( 'B' );
+        teaC = createTrackedEntityAttribute( 'C' );
+        teaD = createTrackedEntityAttribute( 'D' );
+        teaE = createTrackedEntityAttribute( 'E' );
+        manager.save( teaA );
+        manager.save( teaB );
+        manager.save( teaC );
+        manager.save( teaD );
+        manager.save( teaE );
+
+        TrackedEntityType trackedEntityType = createTrackedEntityType( 'A' );
+        manager.save( trackedEntityType );
+
+        TrackedEntityTypeAttribute tetaA = new TrackedEntityTypeAttribute();
+        tetaA.setSearchable( true );
+        tetaA.setTrackedEntityType( trackedEntityType );
+        tetaA.setTrackedEntityAttribute( teaA );
+        manager.save( tetaA );
+        trackedEntityType.getTrackedEntityTypeAttributes().add( tetaA );
+
+        TrackedEntityTypeAttribute tetaB = new TrackedEntityTypeAttribute();
+        tetaB.setSearchable( false );
+        tetaB.setTrackedEntityType( trackedEntityType );
+        tetaB.setTrackedEntityAttribute( teaD );
+        manager.save( tetaB );
+        trackedEntityType.getTrackedEntityTypeAttributes().add( tetaB );
+
+        manager.update( trackedEntityType );
+
+        ProgramTrackedEntityAttribute pteaA = new ProgramTrackedEntityAttribute();
+        pteaA.setSearchable( true );
+        pteaA.setAttribute( teaB );
+        pteaA.setProgram( program );
+        manager.save( pteaA );
+
+        ProgramTrackedEntityAttribute pteaB = new ProgramTrackedEntityAttribute();
+        pteaB.setSearchable( false );
+        pteaB.setAttribute( teaE );
+        pteaB.setProgram( program );
+        manager.save( pteaB );
+
+        program.getProgramAttributes().add( pteaA );
+        program.getProgramAttributes().add( pteaB );
+        manager.update( program );
+
+        tei = createTrackedEntityInstance( orgUnit );
+        tei.setTrackedEntityType( trackedEntityType );
+        manager.save( tei );
+
+        programInstance = new ProgramInstance( program, tei, orgUnit );
+        programInstance.setAutoFields();
+        programInstance.setEnrollmentDate( new Date() );
+        programInstance.setIncidentDate( new Date() );
+        programInstance.setStatus( ProgramStatus.COMPLETED );
+        programInstance.setFollowup( true );
+        manager.save( programInstance );
+    }
+
+    @Test
+    void getIndexableOnlyAttributes()
+    {
+
+        JsonObject json = GET( "/trackedEntityAttributes?indexableOnly=true" ).content( HttpStatus.OK );
+
+        assertFalse( json.isEmpty() );
+
+        Set<String> expectedAttributeSet = new HashSet<>();
+        expectedAttributeSet.add( "\"AttributeA\"" );
```

Comment:
if these tests will only run on java 11 I would use [Set.of](https://docs.oracle.com/javase/9/docs/api/java/util/Set.html#of-E...-)

Reply:
Done. This is only from 2.38 onwards, so Java 11 it is.